metecho/api/serializers.py

```diff
@@ -229,6 +302,12 @@ class EpicSerializer(HashIdModelSerializer):
     branch_url = serializers.SerializerMethodField()
     branch_diff_url = serializers.SerializerMethodField()
     pr_url = serializers.SerializerMethodField()
+    issue = serializers.PrimaryKeyRelatedField(
+        queryset=GitHubIssue.objects.all(),
+        pk_field=serializers.CharField(),
+        required=False,
+        allow_null=True,
+    )
```

Comment:
@jerivas For issues on both Epics and Tasks, I'm wondering what you think about doing a one-way (readonly) nested serialized representation of an attached issue instead of a simple foreign key. That would avoid the front end having to either fetch the full list of all attached issues for an entire project on page-load, or doing lots of individual requests to fetch the attached issue for specific tasks/epics while navigating the site.

Basically, I'm thinking that the front end might store the lists of available and attached tasks in-memory only while the modal is open (not in the Redux store). But we need to display attached issues throughout the site, so as it's written now we'd probably need to fetch all attached issues for a project when we load the project, even if there are multiple pages.

We can talk through this tomorrow if that'd be useful.

Reply:
I think that should work, yes. It's similar to what we did with `TaskSerializer.epic` right?

That being said, in Top Dog we favored filtered lists endpoints instead of nested fields because they play better with DRL. I think it might be worth to move towards that in preparation to the upcoming DRL story. What do you think?

Reply:
Ok, that makes sense -- we can keep them as separate lists.

Maybe I won't worry about fetching the entire list of attached issues and storing it in the redux store. Instead we can fetch on-demand for a particular epic or task if there's one attached, and then only fetch the full per-project list when the modal is opened.